### PR TITLE
ENH: convert error upon writing empty df to warning

### LINKED ISSUE #2234: BUG: Can't write empty GeoJSON
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [x] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
gpd.GeoDataFrame(geometry=[]).to_file("test", driver="GeoJSON")
```
raises
```python
Traceback (most recent call last):

  File "/tmp/ipykernel_2102262/1921918865.py", line 1, in <module>
    gpd.GeoDataFrame(geometry=[]).to_file("test", driver="GeoJSON")

  File "/home/lukas/anaconda3/envs/tf_gpu/lib/python3.9/site-packages/geopandas/geodataframe.py", line 1086, in to_file
    _to_file(self, filename, driver, schema, index, **kwargs)

  File "/home/lukas/anaconda3/envs/tf_gpu/lib/python3.9/site-packages/geopandas/io/file.py", line 304, in _to_file
    schema = infer_schema(df)

  File "/home/lukas/anaconda3/envs/tf_gpu/lib/python3.9/site-packages/geopandas/io/file.py", line 362, in infer_schema
    raise ValueError("Cannot write empty DataFrame to file.")

ValueError: Cannot write empty DataFrame to file.
```

#### Problem description

It's perfectly valid to write an empty GeoJSON. While this may not be true for other filetypes that Geopandas supports, Geopandas shouldn't prevent you from writing an empty GeoJSON.

It was possible, until this PR: https://github.com/geopandas/geopandas/pull/663

Why is this important? When dealing with large amounts of data, it would be a lot of effort to special-case the edge case of an empty collection everywhere that it occurs.

#### Expected Output

By commenting out [these lines](https://github.com/geopandas/geopandas/blob/master/geopandas/io/file.py#L427-L428), Geopandas writes a valid GeoJSON file that looks like this:

```json
{
"type": "FeatureCollection",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
"features": [

]
}
```
This is the expected output.

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]
executable : /home/lukas/anaconda3/envs/tf_gpu/bin/python
machine    : Linux-5.4.0-80-generic-x86_64-with-glibc2.31

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.3.0
GDAL data dir: /home/lukas/anaconda3/envs/tf_gpu/lib/python3.9/site-packages/fiona/gdal_data
PROJ       : 8.1.1
PROJ data dir: /home/lukas/anaconda3/envs/tf_gpu/lib/python3.9/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0
pandas     : 1.3.4
fiona      : 1.8.20
numpy      : 1.19.5
shapely    : 1.8.0
rtree      : 0.9.7
pyproj     : 3.2.1
matplotlib : 3.4.3
mapclassify: None
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : None
pygeos     : 0.10.2

</details>

### PR DESCRIPTION
**Pull request by joooeey**
Since Fiona is able to write empty files for most drivers, Geopandas shouldn't prevent that.

Closes #2234.

I also removed duplicate import statements.

### CODE REVIEW

**Review by martinfleis — APPROVED** [review]
Thanks! Looks okay to me, assuming CI will pass (dev is broken due to changes in shapely, feel free to ignore its failures).

We may want to try adding roundtrips to tests here but I am happy to leave this to fiona. It is quite a corner case after all.

File: geopandas/io/tests/test_file.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def test_to_file_int64(tmpdir, df_points):
 def test_to_file_empty(tmpdir):
     input_empty_df = GeoDataFrame()
```
Comment (martinfleis) [inline geopandas/io/tests/test_file.py:217]:
```suggestion
    input_empty_df = GeoDataFrame(geometry=[])
```

This will fix the CI. But we should probably still raise an error for an empty GeoDataFrame without geometry as that fails. Can you look into that?

Comment (joooeey) [reply geopandas/io/tests/test_file.py:217]: I just had a look at the CI. When trying to write a GeoDataFrame without geometry, you get the pretty descriptive error
```
AttributeError: No geometry data set yet (expected in column 'geometry'.)
```
Hence, I don't think we need to raise an extra error here.

**Review by martinfleis — APPROVED** [review]
Thanks!

**Comment by martinfleis** [comment]
Thanks @joooeey!
